GLOSSARY.md

```diff
@@ -0,0 +1,40 @@
+# GitOps Glossary
+
+This glossary accompanies the [GitOps Principles](./PRINCIPLES.md), and other supporting documents in this repository.
+
+- ## Continuous
+
+    "Continuous" is intended to match the industry standard term: [reconciliation](#reconciliation) continues to happen, not that it must be instantaneous.
+
+- ## Declarative Description
+
+    A configuration that describes the desired operating state of a system without specifying procedures for how that state will be achieved. This separates configuration (the desired state) from the implementation (commands, API calls, scripts etc.) used to achieve that state.
+
+- ## Desired State
+
+    The aggregate of all configuration data needed to start, bootstrap, recreate, run, and recover a system. This does not include data stored within a system, eg. database contents.
+
+- ## Drift
+
+    When a system's actual state changes for any reason other than its [desired state](#desired-state)declarations having changed. When this occurs, the system has drifted from its desired state.
+
+- ## Reconciliation
+
+    The process of ensuring the actual state of a system matches its [desired state](#desired-state) declarations.
+    Contrary to "CIops", any divergence between the two will trigger reconciliation, regardless of where changes occurred.
```

Comment:
Practical systems needn't use triggers in this way -- it's an implementation detail. It is enough to say that reconciliation works continually to make progress towards the desired state.

Reply:
@squaremo this note is here to make clear that doing CD tasks from CI systems in itself is not GitOps, because it is event-based not continuous. I think you're right saying reconciliation is continuous implies that, but since Continuous Integration is so widely known about compared to GitOps CD, I think the word "continuous" in itself needs some further clarification like this to drive home the distinction. Do you think that is very clear without this sentence? Or do you think this sentence should be adjusted to make that point in a different way?

Reply:
> this note is here to make clear that doing CD tasks from CI systems in itself is not GitOps, because it is event-based only, not continuous.

I like @lloydchang 's suggested phrasing. I would elaborate on it a little, by saying that the reasons typical uses of CI fall short of gitops are:

 - action is only taken when a change is made to the source, so it's not continually trying to make progress;
 - events can be processed out of order; e.g., a pipeline fails, then is run again later and succeeds

Reply:
I am not sure progress is the right word here - It will most commonly be associated with progressive delivery which is outside the scope of GitOps either via each progression being a new declaration or through the abstraction of the delivery to another controller - A gitops controller shouldn't be progressively incrementing for example replicas and leaving "attempted" state in between desired and actual.

Reply:
Closing this conversation as it's now out of date with the PR.

Reply:
Reopening conversation so that if desired this can be moved to an issue for feedback between RC 2 and planned v1.0.0. Good points are made here.